third_party/identify_stream_usage/CMakeLists.txt

```diff
@@ -27,22 +27,21 @@ if(GINKGO_BUILD_HIP AND GINKGO_HIP_PLATFORM MATCHES "${HIP_PLATFORM_AMD_REGEX}")
     find_package(hip REQUIRED)
     set_source_files_properties(identify_stream_usage.hip.cpp
                                 test_default_stream_identification.hip.cpp
-                                PROPERTIES HIP_SOURCE_PROPERTY_FORMAT TRUE)
-    hip_add_library(identify_stream_usage_hip
-                    identify_stream_usage.hip.cpp
-                    HIPCC_OPTIONS ${GINKGO_HIPCC_OPTIONS}
-                    CLANG_OPTIONS ${GINKGO_HIP_CLANG_OPTIONS}
-                    NVCC_OPTIONS ${GINKGO_HIP_NVCC_OPTIONS}
-                    SHARED)
-    target_link_libraries(identify_stream_usage_hip PUBLIC hip::amdhip64 ${CMAKE_DL_LIBS})
+                                PROPERTIES LANGUAGE HIP)
+    add_library(identify_stream_usage_hip SHARED identify_stream_usage.hip.cpp)
+    target_link_libraries(identify_stream_usage_hip ${CMAKE_DL_LIBS})
+    set_target_properties(
+        identify_stream_usage_hip
+        PROPERTIES # set target compile options
+                  CXX_STANDARD 17
+                  CXX_STANDARD_REQUIRED ON
+                  POSITION_INDEPENDENT_CODE ON
```

Comment:
This always needs `-fPIC`, because it's a library that will be `LD_PRELOAD`ed. Shared libraries get the flag added automatically, so I'm not 100% sure why this was necessary.